File: appointments/src/main/java/com/medical/homevisits/appointments/appointment/entity/Appointment.java
package com.medical.homevisits.appointments.appointment.entity;

import com.medical.homevisits.appointments.doctor.entity.Doctor;
import com.medical.homevisits.appointments.nurse.entity.Nurse;
import com.medical.homevisits.appointments.patient.entity.Patient;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

import java.sql.Time;
import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Setter
@SuperBuilder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name="appointments")
public class Appointment {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID ID;
    @Enumerated(EnumType.STRING)
    private AppointmentStatus status; //status of the appointment (reserved, available, cancelled or completed)
    private LocalDateTime appointmentStartTime;
    private LocalDateTime appointmentEndTime;
    @ManyToOne
    @JoinColumn(name="doctors", nullable = true)
    private Doctor doctor;
    @ManyToOne
    @JoinColumn(name="nurses", nullable = true)
    private Nurse nurse;
    @ManyToOne
    @JoinColumn(name="patients")
    private Patient patient;
    private String address;
    private String notes;   //available for doctors to add additional info about appointment
}

File: frontend/app/src/main/java/com/medical/wizytydomowe/api/appointments/Appointment.kt
package com.medical.wizytydomowe.api.appointments

import com.medical.wizytydomowe.api.users.Doctor
import com.medical.wizytydomowe.api.users.Patient
import java.io.Serializable

data class Appointment(
    val id: String?,
    val status: String?,
    val appointmentStartTime: String?,
    val appointmentEndTime: String?,
    val doctor: Doctor?,
    val patient: Patient?,
    val address: String?,
    val notes: String?
) : Serializable
